WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -0,0 +1,108 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class BranchAndBound : Selector
+	{
+		private Money _costOfHeader = Money.Satoshis(0);
+		private Money _costPerOutput = Money.Satoshis(0);
+		private int _bnbTryLimit = 5;
+		private Random _random = new();
+
+		private Money[]? UtxoSorted { get; set; }
+
+		public bool TryGetExactMatch(Money target, List<Money> availableCoins, out List<Money> selectedCoins)
```

Comment:
Typical idiom is:

```cs
public bool TryGetExactMatch(Money target, [NotNullWhen(true)] out List<Money>? selectedCoins)
```

Reply:
What does this do?

Reply:
Well, it helps compiler to know that if you do:

```cs
if (c.TryGetExactMatch(target, out List<Money>? selectedCoins)) {
   // here it won't bother you warning you that `selectedCoins` can be null because you have added `NotNullWhen` that says that: `selectedCoins` cannot be `null` if `TryGetExactMatch` returns true
}